### DISCUSSION TITLE
Transaction with dynamic post_max_size

### DISCUSSION THREAD

**Discussion by clpir3s** [reactphp/reactphp#558] [Q&A] [answered]
Hi all,

I have a requirement for a download content that exceed the max request defined 16MB.
I was trying to create a subclass to override this value, but I notice that this is not possible.

There is a way to override this limitation? 
https://github.com/reactphp/http/blob/1.x/src/Io/Transaction.php#L167

**Comment by clpir3s** [comment]
btw, 

I already create a local class of Browser and new method that allow me to change this value.

**Reply by SimonFrings** [reply, marked as answer]
Hey @clpir3s, if you're referring to the [`withResponseBuffer()` method ](https://github.com/reactphp/http#withresponsebuffer), you're absolutely right as this method can be used to change the maximum size for buffering a response body.

```php
$browser = $browser->withResponseBuffer(1024 * 1024);
```

It's great that you've already found an answer to your question, but please provide a more detailed explanation of your findings next time, so that others with the same question can easily find the answer here as well 😁 :+1:

**Comment by clpir3s** [comment]
Hi @SimonFrings,

many thanks I didn't see this method.

sure!

I implement a method that is already exists  😁 😁 😁
👍
